node-test/src/test/scala/org/bitcoins/node/UpdateBloomFilterTest.scala

```diff
@@ -0,0 +1,190 @@
+package org.bitcoins.node
+
+import org.bitcoins.testkit.wallet.BitcoinSWalletTest
+import org.scalatest.FutureOutcome
+import org.bitcoins.node.models.Peer
+import org.bitcoins.chain.models.BlockHeaderDAO
+import org.bitcoins.chain.blockchain.ChainHandler
+import org.bitcoins.testkit.node.NodeTestUtil
+import org.bitcoins.chain.config.ChainAppConfig
+import org.bitcoins.node.config.NodeAppConfig
+import org.bitcoins.node.networking.peer.DataMessageHandler
+import org.bitcoins.core.protocol.transaction.Transaction
+import scala.concurrent._
+import scala.concurrent.duration._
+import org.scalatest.compatible.Assertion
+import org.bitcoins.core.currency._
+import scala.util.Try
+import akka.actor.Cancellable
+import org.scalatest.run
+import org.scalatest.exceptions.TestFailedException
+import org.bitcoins.core.wallet.fee.SatoshisPerByte
+
+class UpdateBloomFilterTest extends BitcoinSWalletTest {
+  override type FixtureParam = WalletWithBitcoind
+
+  def withFixture(test: OneArgAsyncTest): FutureOutcome =
+    withFundedWalletAndBitcoind(test)
+
+  it must "update the bloom filter with an address" in { param =>
+    val WalletWithBitcoind(wallet, rpc) = param
+    implicit val chainConf: ChainAppConfig = config
+    implicit val nodeConf: NodeAppConfig = config
+
+    val assertionP = Promise[Assertion]
+    val assertionF = assertionP.future
+
+    // we want to schedule a runnable that aborts
+    // the test after a timeout, but then
+    // we need to cancel that runnable once
+    // we get a result
+    var cancelable: Option[Cancellable] = None
+    val timeout = 15.seconds
+
+    for {
+      _ <- config.initialize()
+
+      firstBloom <- wallet.getBloomFilter()
+
+      // this has to be generated after our bloom filter
+      // is calculated
+      addressFromWallet <- wallet.getNewAddress()
+      _ = logger.info(s"Address from wallet: $addressFromWallet")
+
+      spv <- {
```

Comment:
we need a bunch of `val`s found in that block to construct the node, `rpc`, `cancellable`, `assertionP`, `addressfromWallet`. So that's a bit hard... But extracted some lines into a method